pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -573,10 +564,13 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard, ICo
         // is achieved efficiently by simply updating the sum of the denormalized weights.
         _denormWeightSum = weightSumAfterAdd;
 
-        // Finally, we store the new token's weight.
-        // `_encodeTokenState` performs an external call to `token` (to get its decimals), but this is
-        // reentrancy safe since that is a static call.
-        _tokenState[token] = _encodeTokenState(token, normalizedWeight, normalizedWeight, weightSumAfterAdd);
+        // Finally, we store the new token's weight and scaling factor.
+        _tokenState[token] = ManagedPoolTokenLib.setTokenWeight(
+            ManagedPoolTokenLib.setTokenScalingFactor(bytes32(0), token),
```

Comment:
Sneakily initializing the state to include the scaling factor is rather odd - I had to spend about 3 minutes going back and forth trying to understand what this line was doing.

An issue here is that there's no guarantee the full state is scaling factor + weight. We should perhaps have a function that returns a fully initialied state, so we can be sure we're not forgetting anything.

Reply:
[95bcb3b](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1730/commits/95bcb3bac511e86250b6a9862572c5ce0b6661ae)